cooked-validators/src/Cooked/Tx/Constraints/Type.hs

```diff
@@ -62,6 +74,7 @@ type SpendsConstrs a =
 -- transaction, and miscellaneous constraints 'MiscConstraint' including input
 -- constraints, time constraints, or signature requirements.
 data Constraints = [MiscConstraint] :=>: [OutConstraint]
+  deriving (Eq)
```

Comment:
I think that it's good (maybe even expected) that the order of constraints matters. At least to me, the syntax `==` suggests that.